service/featuregate/gates.go

```diff
@@ -57,6 +58,21 @@ func (r *Registry) Apply(cfg map[string]bool) {
 	}
 }
 
+// MustApply a configuration in the form of a map of Gate identifiers to boolean values.
+// Sets only those values provided in the map, other gate values are not changed.
+func (r *Registry) MustApply(cfg map[string]bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	for id, val := range cfg {
+		if g, ok := r.gates[id]; ok {
+			g.Enabled = val
+			r.gates[g.ID] = g
+		} else {
+			panic(fmt.Sprintf("feature gate %s is unregistered", id))
```

Comment:
Thoughts on returning an error here instead of calling `panic`? It's what @mx-psi's original suggestion was.

Reply:
@mx-psi You are right, I follow the example 2.

Reply:
Got it. Thanks for clarifying!